src/infra/core/src/remote_alias_resolver_impl.rs

```diff
@@ -0,0 +1,308 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::str::FromStr;
+use std::sync::Arc;
+
+use auth::OdfServerAccessTokenResolver;
+use dill::*;
+use internal_error::{InternalError, ResultIntoInternal};
+use kamu_core::*;
+use opendatafabric::{self as odf, DatasetRefRemote};
+use url::Url;
+
+use crate::UrlExt;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+pub struct RemoteAliasResolverImpl {
```

Comment:
Note, there are 2 types of remote aliases: push & pull.
You've named everything every general, but only work with push aliases.
The naming of either the component, or particular methods should be improved.

Reply:
I've already made some preparation where service accept Alias kind to support both push and pull. And once we will be ready to improve pull flow it will be reused